File: 3189-find-champion-ii/find-champion-ii.java
class Solution {
    public int findChampion(int n, int[][] edges) {
        int countParent[] = new int[n];
        Arrays.fill(countParent, 0);

        for (int[] edge: edges)
            countParent[edge[1]] += 1;
        
        List<Integer> champions = new ArrayList<Integer>();

        for (int i = 0; i < n; i++)
            if (countParent[i] == 0)
                champions.add(i);

        if (champions.size() > 1)
            return -1;
        return champions.get(0); 
    }
}

File: 3189-find-champion-ii/find-champion-ii.py
class Solution:
    def findChampion(self, n: int, edges: List[List[int]]) -> int:
        countParents = [0] * n
        for edge in edges:
            countParents[edge[1]] += 1
        
        champions = []
        for i, c in enumerate(countParents):
            if not c:
                champions.append(i)

        if len(champions) > 1:
            return -1
        return champions[0]